central/complianceoperator/v2/checkresults/datastore/datastore_impl.go

```diff
@@ -209,6 +211,21 @@ func (d *datastoreImpl) ComplianceProfileResults(ctx context.Context, query *v1.
 		return nil, err
 	}
 
+	var ruleNames []string
```

Comment:
The pattern we've been using for data we have to go back and retrieve post query is to do that in the service.

Reply:
As such I would probably process a single rule at a time as I was building the results as well.  Here is one example of where we go back and get rule information after the fact.

https://github.com/stackrox/stackrox/blob/78463b3641c8d863787869dc3c111a6ef00e88ca/central/complianceoperator/v2/checkresults/service/service_impl.go#L316